gitlab/client.py

```diff
@@ -931,7 +931,7 @@ def http_list(
 
         if iterator:
```

Comment:
To fix the regression and replicate as closely as things were before the change. I think this should be the change. But maybe I'm wrong.

```suggestion
        if iterator and page is None:
```

Agree with @nejch on adding a unit-test for this.

Reply:
I see. Indeed, the previous code would return a list if `page` is used, even if `is_false` is `True`, but should it return a generator instead?

Reply:
I see. Let's restore to the old behavior than. I'll update the MR and add a note on the docs.

Reply:
@nejch I agree. Thanks.